src/v/compat/id_allocator_generator.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#pragma once
+
+#include "cluster/errc.h"
+#include "cluster/id_allocator.h"
+#include "cluster/types.h"
+#include "compat/generator.h"
+#include "model/timeout_clock.h"
+#include "random/generators.h"
+
+#include <chrono>
+#include <cstdint>
+#include <limits>
+#include <type_traits>
+
+namespace compat {
+
+template<>
+struct instance_generator<cluster::allocate_id_request> {
+    static cluster::allocate_id_request random() {
+        return cluster::allocate_id_request{model::timeout_clock::duration(
```

Comment:
would it be easier to have a helper to gen durations? And any value that its underlying type could hold would be a valid generated value?

Reply:
Yeah, might be worth putting this in compat/generator.h. Also I'm a bit confused about the mismatch between `chrono::milliseconds` and `timeout_clock::duration`, but would it make sense to generate a value between `milliseconds::min()` and `milliseconds::max()`?